src/components/PinInput/src/PinInputControl.vue

```diff
@@ -1,51 +1,65 @@
 <template>
 	<div
-		:class="{
-			[$s.PinInputContainer]: true,
-			[$s.shake]: isShaking,
-			[$s.disabled]: disabled,
-			[$s.error]: Boolean($slots.error),
-		}"
+		:class="$s.PinInputContainer"
+		:style="computedStyles"
 	>
 		<input
-			v-for="(val, i) in pin"
-			:key="i"
-			:ref="getPinCellRef(i)"
-			:value="val"
-			:autocomplete="i === 0 ? 'one-time-code' : 'off'"
-			:disabled="disabled"
-			:maxlength="i === 0 ? pinLength : 1"
-			:class="{
-				[$s.PinInputCell]: true,
-				[$s.fill]: variant === 'fill',
-				[$s.error]: invalid,
-			}"
+			ref="input"
+			:class="$s.PinInput"
+			:maxlength="pinLength"
+			:value="inputValue"
 			type="text"
 			inputmode="numeric"
 			pattern="[0-9]*"
-			required
-			@input="onInputPin($event, i)"
-			@paste="onPastePin($event, i)"
-			@focus="onFocusPin($event, i)"
-			@keydown.delete="onDelete($event, i)"
+			:disabled="disabled || isShaking"
+			@keypress="sanitizePinInput"
+			@keyup="updateCaretPosition"
+			@input="onInputPin"
+			@focus="setFocus(true)"
+			@blur="setFocus(false)"
 		>
+
+		<div
+			:class="{
+				[$s.PinInputCells]: true,
+				[$s.isShaking]: isShaking,
+			}"
+		>
+			<div
+				v-for="pinPosition in pinLength"
+				:key="pinPosition"
+				:class="{
+					[$s.PinInputCell]: true,
+					[$s.isFocused]: isCellFocused(pinPosition - 1),
+					[$s.fill]: variant === 'fill',
+					[$s.error]: invalid,
+				}"
+			>
+				{{ inputValue[pinPosition - 1] }}
+			</div>
+		</div>
 	</div>
 </template>
 
 <script>
-import { BASE_TEN } from '@square/maker/utils/constants';
+const INPUT_FILTER_REGEX = /\D/gi;
+const INPUT_KEY_REGEX = /\d/;
 
 const DEFAULT_INPUT_SIZE = 6;
+const PIN_CELL_WIDTH = 50;
+const PIN_CELL_SPACING = 8;
+const ONE = 1;
+const COMPLETE_TIMEOUT_MS = 500;
 
 export default {
 	props: {
 		/**
-		 * Lenth of pin
+		 * Length of pin
 		 */
 		pinLength: {
 			type: Number,
 			default: DEFAULT_INPUT_SIZE,
-			validator: (value) => value > 0,
+			validator: (value) => value > 0 && value <= DEFAULT_INPUT_SIZE,
```

Comment:
oh, interesting, is there a reason why we're now limiting the max pin input size to 6? I'm fine either having a limit or not, just curious why you added this.

Reply:
Let me try with larger sizes and see what breaks.

Reply:
Reverted